mantaray/gui.py

```diff
@@ -168,15 +165,18 @@ def text(self) -> str:
         return self.get_current_view().textwidget.get("1.0", "end - 1 char")
 
     def get_server_views(self) -> list[ServerView]:
-        return [
-            view for view in self.views_by_id.values() if isinstance(view, ServerView)
+        server_views = [
+            v for v in self.views_by_id.values() if isinstance(v, ServerView)
         ]
+        server_views.sort(key=(lambda v: self.view_selector.index(v.view_id)))
+        return server_views
 
     def _show_change_nick_dialog(self) -> None:
-        core = self.get_current_view().server_view.core
-        new_nick = ask_new_nick(self.winfo_toplevel(), core.nick)
-        if new_nick != core.nick:
-            core.send(f"NICK {new_nick}")
+        server_view = self.get_current_view().server_view
+        new_nick = ask_new_nick(self.winfo_toplevel(), server_view.core.nick)
+        if new_nick != server_view.core.nick:
+            server_view.core.send(f"NICK {new_nick}")
+            server_view.settings.nick = new_nick
```

Comment:
Presumably the nick should be saved to settings only when it changes successfully.